java/statestore/src/test/java/sleeper/statestore/transactionlog/TransactionLogSnapshotCreatorIT.java

```diff
@@ -0,0 +1,397 @@
+        TableProperties table = createTable("test-table-id-1", "test-table-1");
+        StateStore stateStore = createStateStoreWithInMemoryTransactionLog(table);
+        stateStore.initialise();
+        FileReferenceFactory factory = FileReferenceFactory.from(stateStore);
+        stateStore.addFile(factory.rootFile(123L));
+        runSnapshotCreator(table);
+
+        // When
+        runSnapshotCreator(table);
+
+        // Then
+        assertThat(snapshotStore(table).getLatestSnapshots())
+                .isEqualTo(new LatestSnapshots(
+                        filesSnapshot(table, 1),
+                        partitionsSnapshot(table, 1)));
+    }
+
+    @Test
+    void shouldNotCreateSnapshotForTableWithNoTransactions() throws Exception {
+        // Given
+        TableProperties table = createTable("test-table-id-1", "test-table-1");
+
+        // When
+        runSnapshotCreator(table);
+
+        // Then
+        assertThat(snapshotStore(table).getLatestSnapshots())
+                .isEqualTo(LatestSnapshots.empty());
+        assertThat(snapshotStore(table).getFilesSnapshots()).isEmpty();
+        assertThat(snapshotStore(table).getPartitionsSnapshots()).isEmpty();
+    }
+
+    @Test
+    void shouldNotCreateFileSnapshotForTableWithOnlyPartitionTransactions() throws Exception {
+        // Given
+        TableProperties table = createTable("test-table-id-1", "test-table-1");
+        StateStore stateStore = createStateStoreWithInMemoryTransactionLog(table);
+        stateStore.initialise();
+
+        // When
+        runSnapshotCreator(table);
+
+        // Then
+        assertThat(snapshotStore(table).getLatestSnapshots())
+                .isEqualTo(new LatestSnapshots(null, partitionsSnapshot(table, 1)));
+        assertThat(snapshotStore(table).getFilesSnapshots()).isEmpty();
+        assertThat(snapshotStore(table).getPartitionsSnapshots()).containsExactly(partitionsSnapshot(table, 1));
+    }
+
+    @Test
+    void shouldRemoveSnapshotFilesIfDynamoTransactionFailed() throws Exception {
+        // Given
+        TableProperties table = createTable("test-table-id-1", "test-table-1");
+        StateStore stateStore = createStateStoreWithInMemoryTransactionLog(table);
+        stateStore.initialise();
+        FileReferenceFactory factory = FileReferenceFactory.from(stateStore);
+        stateStore.addFile(factory.rootFile(123L));
+
+        // When / Then
+        DuplicateSnapshotException exception = new DuplicateSnapshotException("test.parquet", new Exception());
+        assertThatThrownBy(() -> runSnapshotCreator(table, failedUpdate(exception)))
+                .isInstanceOf(RuntimeException.class)
+                .hasCause(exception);
+        assertThat(snapshotStore(table).getLatestSnapshots())
+                .isEqualTo(LatestSnapshots.empty());
+        assertThat(Files.exists(filesSnapshotPath(table, 1))).isFalse();
+        assertThat(Files.exists(partitionsSnapshotPath(table, 1))).isFalse();
```

Comment:
This is only testing whichever is the first snapshot to be saved to DynamoDB, because they're saved in separate calls. It looks like this needs to be two tests.